File: SmartBooks.Domains/Entities/Warehouse.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SmartBooks.Domains.Entities
{
    public class Warehouse : AppBaseEntity
    {
        public Warehouse()
        {
            Bins = new HashSet<Bin>();
        }
        [StringLength(50)]
        public string Code { get; set; }
        [StringLength(150)]
        public string Description { get; set; }
        public int DefaultReceivingBin { get; set; }
        public int DefaultDespatchBin { get; set; }

        public ICollection<Bin> Bins { get; set; }
    }
}

       

File: SmartBooksFrontend/src/app/models/warehouse.ts
export interface Warehouse {
  id: number;
  code: string;
  description: string;
  defaultReceivingBin: number;
  defaultDespatchBin: number;
  bins: Array<Bin>;
}

export interface Bin {
  id: number;
  warehouseId: number;
  code: string;
  description: string;
}
